File: diploma-server/src/main/java/org/arhor/diploma/config/ScheduledTasks.java
package org.arhor.diploma.config;

import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Configuration;

@Slf4j
@Configuration
public class ScheduledTasks {

//  /** Scheduled task to clear cache `users` every 5 minutes. */
//  @CacheEvict(allEntries = true, value = { CACHE_USERS })
//  @Scheduled(fixedDelay = 5 * 60 * 1000, initialDelay = 500)
//  public void reportUsersCacheEvict() {
//    log.info("Flush `users` cache");
//  }
//
//  /** Scheduled task to clear caches `langs` and `roles` every 24 hours. */
//  @CacheEvict(allEntries = true, value = { CACHE_LANGS, CACHE_ROLES })
//  @Scheduled(fixedDelay = 24 * 60 * 60 * 1000, initialDelay = 500)
//  public void reportLangsRolesCacheEvict() {
//    log.info("Flush `langs` and `roles` caches");
//  }
}

File: app-server/module-main/src/main/kotlin/com/github/arhor/linden/dragon/tavern/service/schedule/ScheduledTasks.kt
package com.github.arhor.linden.dragon.tavern.service.schedule;

import com.github.arhor.linden.dragon.tavern.common.time.MINUTE
import com.github.arhor.linden.dragon.tavern.data.persistence.domain.Account
import mu.KLogging
import org.springframework.cache.annotation.CacheEvict
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class ScheduledTasks {

    /**
     * Scheduled task to clear account caches every 5 minutes.
     */
    @CacheEvict(allEntries = true, value = [Account.CACHE])
    @Scheduled(fixedDelay = 5 * MINUTE, initialDelay = 5 * MINUTE)
    fun accountCacheEvict() {
        logger.debug { "Flush [${Account.CACHE}] cache(s)" }
    }

    companion object : KLogging()
}
